app/src/main/java/com/example/habitsmasher/listeners/SwipeListener.java

```diff
@@ -0,0 +1,46 @@
+package com.example.habitsmasher.listeners;
+
+import com.example.habitsmasher.Habit;
+import com.example.habitsmasher.HabitList;
+import com.example.habitsmasher.R;
+import com.example.habitsmasher.User;
+import com.example.habitsmasher.ui.dashboard.EditHabitFragment;
+import com.example.habitsmasher.ui.dashboard.HabitItemAdapter;
+import com.example.habitsmasher.ui.dashboard.HabitListFragment;
+import com.example.habitsmasher.ui.dashboard.RecyclerTouchListener;
+/**
+ * This is the SwipeListener class
+ * It implements the RecyclerTouchListener.OnSwipeOptionsClickListener
+ * Its purpose is to perform the onSwipeOptionClicked listener actions
+ */
+public class SwipeListener implements RecyclerTouchListener.OnSwipeOptionsClickListener{
+    private HabitItemAdapter _habitItemAdapter;
+    private HabitListFragment _fragment;
+    private HabitList _habitList;
+    private User _user;
+
+    public SwipeListener(HabitItemAdapter habitItemAdapter, HabitListFragment fragment, HabitList habitList, User user) {
+        _habitItemAdapter = habitItemAdapter;
+        _fragment = fragment;
+        _habitList = habitList;
+        _user = user;
+    }
+
+    @Override
+    public void onSwipeOptionClicked(int viewID, int position) {
+        switch (viewID){
+            // if edit clicked
+            case R.id.edit_button:
+                EditHabitFragment editHabitFragment = new EditHabitFragment(position,
+                        _habitItemAdapter._snapshots.get(position),
+                        _fragment);
+                editHabitFragment.show(_fragment.getFragmentManager(), "Edit Habit");
+                break;
+            // if delete clicked
+            case R.id.delete_button:
+                Habit habitToDelete = _habitItemAdapter._snapshots.get(position);
+                _habitList.deleteHabit(_fragment.getActivity(), _user.getUsername(), habitToDelete, position);
+                break;
+        }
```

Comment:
Since we already pass in an instance of HabitListFragment into this listener, we should probably just call the methods openEditDialogBox() and updateListAfterDelete() in the HabitListFragment class here instead of repeating the same logic.

- [ ] Replace logic with calls to method in HabitListFragment

Reply:
This'll reduce the number of fields you'll have to pass in also, as the user, habit list and adapter are used in the methods defined in HabitListFragment.